test/cmd/newapp.sh

```diff
@@ -62,6 +62,8 @@ os::cmd::expect_success_and_text 'oc new-app ruby-helloworld-sample -o yaml' 'MY
 os::cmd::expect_success_and_text 'oc new-app ruby-helloworld-sample -o yaml' 'ADMIN_USERNAME'
 os::cmd::expect_success_and_text 'oc new-app ruby-helloworld-sample -o yaml' 'ADMIN_PASSWORD'
 os::cmd::expect_success_and_text 'oc new-app ruby-helloworld-sample --param MYSQL_PASSWORD=hello -o yaml' 'hello'
+os::cmd::expect_success_and_text  'oc new-app -e FOO=BAR -f examples/jenkins/jenkins-ephemeral-template.json -o json | grep FOO | wc -l' '1'
+os::cmd::expect_success_and_text  'oc new-app -e OPENSHIFT_ENABLE_OAUTH=false -f examples/jenkins/jenkins-ephemeral-template.json -o json | grep -A 1 OPENSHIFT_ENABLE_OAUTH | grep false | wc -l' '1'
```

Comment:
I think a better approach here would be (wrapped in the correct `os::cmd` invocations):

```
oc new-app -e OPENSHIFT_ENABLE_OAUTH=false -f examples/jenkins/jenkins-ephemeral-template.json
oc get <whatever_object> -o jsonpath=<path_to_environment_field>
```

Then you could directly check that whatever part of whatever object you are creating has the field you need. The `grep -A1` check is understandably very fragile.

If you want to do counting, etc, use a Golang template.